flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/utils/ExecNodeMetadataUtil.java

```diff
@@ -198,8 +205,9 @@ public static Class<? extends ExecNode<?>> retrieveExecNode(String name, int ver
     }
 
     public static <T extends ExecNode<?>> boolean isUnsupported(Class<T> execNode) {
-        return !StreamExecNode.class.isAssignableFrom(execNode)
-                || UNSUPPORTED_JSON_SERDE_CLASSES.contains(execNode);
+        boolean streamOrKnownExecNode =
+                StreamExecNode.class.isAssignableFrom(execNode) || execNodes().contains(execNode);
+        return !streamOrKnownExecNode || UNSUPPORTED_JSON_SERDE_CLASSES.contains(execNode);
```

Comment:
Out of curiosity: all these maps/sets are intentionally mutable?

Reply:
Good question.  Looks like that was just an initial decision from the first PRs setting this.  

If you'd like, I'd be happy to make it immutable in a separate PR/commit.  I may want to defer that work to later since I have intermediate work using the current approach.

Reply:
yep, it could be done separately